pkg/grpc/universalapi/api_workflow_test.go

```diff
@@ -0,0 +1,156 @@
+		},
+		{
+			testName:          "No workflow component provided in pause request",
+			apiToBetested:     PauseWorkflow,
+			instanceId:        "inst1",
+			workflowComponent: "",
+			errorExcepted:     true,
+			expectedError:     messages.ErrNoOrMissingWorkflowComponent,
+		},
+		{
+			testName:          "workflow component does not exist in pause request",
+			apiToBetested:     PauseWorkflow,
+			instanceId:        "inst1",
+			workflowComponent: "fakeWorkflowNotExist",
+			errorExcepted:     true,
+			expectedError:     messages.ErrWorkflowComponentDoesNotExist.WithFormat("fakeWorkflowNotExist"),
+		},
+		{
+			testName:          "pause for this instance throws error",
+			apiToBetested:     PauseWorkflow,
+			instanceId:        "errorInstanceId",
+			workflowComponent: "fakeWorkflow",
+			errorExcepted:     true,
+			expectedError:     messages.ErrPauseWorkflow.WithFormat("errorInstanceId"),
+		},
+		{
+			testName:          "All is well in pause request",
+			apiToBetested:     PauseWorkflow,
+			instanceId:        "instanceId1",
+			workflowComponent: "fakeWorkflow",
+			errorExcepted:     false,
+		},
+		{
+			testName:          "No instance id present in resume request",
+			apiToBetested:     ResumeWorkflow,
+			instanceId:        "",
+			workflowComponent: "fakeWorkflow",
+			errorExcepted:     true,
+			expectedError:     messages.ErrMissingOrEmptyInstance,
+		},
+		{
+			testName:          "No workflow component provided in resume request",
+			apiToBetested:     ResumeWorkflow,
+			instanceId:        "inst1",
+			workflowComponent: "",
+			errorExcepted:     true,
+			expectedError:     messages.ErrNoOrMissingWorkflowComponent,
+		},
+		{
+			testName:          "workflow component does not exist in resume request",
+			apiToBetested:     ResumeWorkflow,
+			instanceId:        "inst1",
+			workflowComponent: "fakeWorkflowNotExist",
+			errorExcepted:     true,
+			expectedError:     messages.ErrWorkflowComponentDoesNotExist.WithFormat("fakeWorkflowNotExist"),
+		},
+		{
+			testName:          "resume for this instance throws error",
+			apiToBetested:     ResumeWorkflow,
+			instanceId:        "errorInstanceId",
+			workflowComponent: "fakeWorkflow",
+			errorExcepted:     true,
+			expectedError:     messages.ErrResumeWorkflow.WithFormat("errorInstanceId"),
+		},
+		{
+			testName:          "All is well in resume request",
+			apiToBetested:     ResumeWorkflow,
+			instanceId:        "instanceId1",
+			workflowComponent: "fakeWorkflow",
+			errorExcepted:     false,
+		},
+	}
+
+	// Setup universal dapr API
+	fakeAPI := &UniversalAPI{
+		Logger:             testLogger,
+		Resiliency:         resiliency.New(nil),
+		WorkflowComponents: fakeWorkflows,
+	}
+
+	for _, tt := range testCases {
+		t.Run(tt.testName, func(t *testing.T) {
+			req := &runtimev1pb.WorkflowActivityRequest{
+				InstanceId:        tt.instanceId,
+				WorkflowComponent: tt.workflowComponent,
+			}
+			var err error
+			if tt.apiToBetested == PauseWorkflow {
```

Comment:
Can you test Terminate too? Since it's pretty much the same

Reply:
Done